Tests/js/views/subitem/assets/ez-subitemgridview-tests.js

```diff
@@ -597,6 +598,70 @@ YUI.add('ez-subitemgridview-tests', function (Y) {
         },
     });
 
+    //Regression test for EZP-25671
+    gridItemsDoNotDuplicateTest = new Y.Test.Case({
+        name: "eZ Subitem Grid View grid items do not duplicates test",
+
+        setUp: function () {
+            this.location = new Y.Model({
+                childCount: 5,
+            });
+            this.view = new Y.eZ.SubitemGridView({
+                location: this.location,
+                container: '.container',
+            });
+            this.view.render();
+            this.view.set('active', true);
+        },
+
+        tearDown: function () {
+            this.view.destroy();
+        },
+
+        _getSubItemStruct: function (baseId) {
+            return {
+                content: new Y.Model({id: 'content-' + baseId}),
+                location: new Y.Model({id: 'location-' + baseId}),
+                contentType: new Y.Model({id: 'contentType-' + baseId}),
+            };
+        },
+
+        "Should NOT duplicates item": function () {
+            var subitems = [
+                    this._getSubItemStruct(1),
+                    this._getSubItemStruct(2),
+                    this._getSubItemStruct(3),
+                    this._getSubItemStruct(4),
+                    this._getSubItemStruct(5),
+                    this._getSubItemStruct(6),
+                    this._getSubItemStruct(7),
+                    this._getSubItemStruct(8),
+                    this._getSubItemStruct(9),
+                    this._getSubItemStruct(10)
+                ],
+                container = this.view.get('container'),
+                gridItems;
+
+            this.view.set('subitems', subitems);
+            gridItems = container.all('.ez-subitemgrid-item');
+            Assert.areEqual(
+                subitems.length,
+                gridItems.size(),
+                "There should be one grid item per subitem"
+            );
+
+            this.view.set('subitems', this._getSubItemStruct(11));
```

Comment:
this is working because you are lucky (or because `subitems` should have a validator), but the `subitems` is supposed to accept only arrays. So this should be:
```js
this.view.set('subitems', [this._getSubItemStruct(11)]);
```